store.go

```diff
@@ -2934,11 +2978,39 @@ func (s *store) Mounted(id string) (int, error) {
 }
 
 func (s *store) UnmountImage(id string, force bool) (bool, error) {
-	img, err := s.Image(id)
+	lstore, err := s.getLayerStore()
+	if err != nil {
+		return false, err
+	}
+	if err := lstore.startWriting(); err != nil {
+		return false, err
+	}
+	defer lstore.stopWriting()
+
+	// images can be mounted only in the rw-store, so don't
+	// check in the additional stores.
```

Comment:
I thought image mounts are restricted to the primary _layer_ store.

AFAICS `MountImage` would now be perfectly content with finding an image in a RO _image_ store. (How would that happen? Start with an ~empty RO image store, pull an image to primary store, untag it without deleting layers, then update the RO store to include the same image. Should unmounting such an RO-store-tagged-only image be expected to work? Meh, I don’t know, but when mounting does work…)

---

(IF _both_ layer and image stores were restricted to the RW store, that would be best commented at the start of the function.)